src/python/impactx/MADXParser.py

```diff
@@ -0,0 +1,470 @@
+
+        """
+        return string.replace(" ", "")
+
+    def __str__(self):
+
+        if self.__lattice:
+            length = 0.0
+
+            # drift, dipole, quadrupole, dipedge
+            nTypes = [0, 0, 0, 0]
+
+            for elem in self.__lattice["elem"]:
+                for e in self.__elements:
+                    if elem == e["name"]:
+                        length += e["l"]
+
+                        if e["type"] == "drift":
+                            nTypes[0] += 1
+                        elif e["type"] == "sbend":
+                            nTypes[1] += 1
+                        elif e["type"] == "quadrupole":
+                            nTypes[2] += 1
+                        elif e["type"] == "dipedge":
+                            nTypes[3] += 1
+                        break
+
+            sign = "*" * 70
+            info = (
+                sign
+                + "\n"
+                + "MADX-Parser information:\n"
+                + "         length:\t"
+                + str(length)
+                + " [m]\n"
+                + "      #elements:\t"
+                + str(len(self.__lattice["elem"]))
+                + "\n"
+                + "            *       #drift:\t"
+                + str(nTypes[0])
+                + "\n"
+                + "            *      #dipole:\t"
+                + str(nTypes[1])
+                + "\n"
+                + "            *  #quadrupole:\t"
+                + str(nTypes[2])
+                + "\n"
+                + "            *     #dipedge:\t"
+                + str(nTypes[3])
+                + "\n"
+                + "           beam:\t\n"
+                + "            *     particle:\t"
+                + self.beam["particle"]
+                + "\n"
+                + "            * total energy:\t"
+                + str(self.beam["energy"])
+                + " [GeV]\n"
+                + sign
+            )
+
+            return info
+        else:
+            return "No information available."
+
+    def getBeamline(self):
+        if self.__lattice:
+
+            beamline = []
+
+            for elem in self.__lattice["elem"]:
+                for e in self.__elements:
+                    if elem == e["name"]:
+                        if e["type"] == "drift":
+                            print("Drift L= " + str(e["l"]))
+                            beamline.append(e)
+                        elif e["type"] == "sbend":
+                            print("Sbend L= ", e["l"], " angle = ", e["angle"])
+                            beamline.append(e)
+                        elif e["type"] == "quadrupole":
+                            print("Quadrupole L= ", e["l"], " k1 = ", e["k1"])
+                            beamline.append(e)
+                        elif e["type"] == "dipedge":
+                            print("Dipedge H= ", e["h"], " E1 = ", e["e1"])
```

Comment:
Leftover debug print statements :)

Reply:
Hm, it's there in Andreas' original version, too. Should we completely remove it or make it optional? Otherwise, it will print every beamline element once (however, it's still incomplete, right now).

Reply:
please remove, this is debug stuff

Reply:
```suggestion
                            # print("Drift L= " + str(e["l"]))
                            beamline.append(e)
                        elif e["type"] == "sbend":
                            # print("Sbend L= ", e["l"], " angle = ", e["angle"])
                            beamline.append(e)
                        elif e["type"] == "quadrupole":
                            # print("Quadrupole L= ", e["l"], " k1 = ", e["k1"])
                            beamline.append(e)
                        elif e["type"] == "dipedge":
                            # print("Dipedge H= ", e["h"], " E1 = ", e["e1"])
```